### PR TITLE
Fix bad https redirections in Nginx

### PR DESCRIPTION
**Pull request by n1mh**
Gevent doesn't change from `http` to `https` protocol and, in case you configure NGINX to use only secure server, you'll get frontpage without styles sheets.

This PR makes `gevent` to use the enviroment variables to know if the protocol is `http` or `https`.

### CODE REVIEW

File: gecoscc/socks.py [hunk 1]
```diff
@@ -111,6 +111,15 @@ def delete_computer(object_id, path):
     }))
 
 
+class GecosWSGIHandler(GunicornWSGIHandler):
+
+    def get_environ(self):
+        env = super(GecosWSGIHandler, self).get_environ()
+        for key, value in self._headers():
```
Comment (alfem) [inline gecoscc/socks.py]: Please refactor this code. A full loop is not needed.

**Comment by alfem** [comment]
"HTTP_X_FORWARDED_PROTO" must be set in nginx (or in a proxy) or this 'trick' will not work.

**Comment by n1mh** [comment]
Hello @alfem.

HTTP_X_FORWARDED_PROTO is a very common configuration in `nginx` when you are using it as a reverse proxy. For example:

~~~
  location / {
      proxy_set_header Host $host;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_pass_request_headers on;
      proxy_redirect off;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_pass http://@app;
      proxy_http_version 1.1;
  }
~~~

Actually, `gevent` uses the same idea (get from `/opt/gecosccui-2.2.0/lib/python2.7/site-packages/gunicorn/http/wsgi.py`). Check out lines 104-106 and 123:

~~~
 97     for hdr_name, hdr_value in req.headers:
 98         if hdr_name == "EXPECT":
 99             # handle expect
100             if hdr_value.lower() == "100-continue":
101                 sock.send(b"HTTP/1.1 100 Continue\r\n\r\n")
102         elif x_forwarded_for_header and hdr_name == x_forwarded_for_header:
103             forward = hdr_value
104         elif secure_headers and (hdr_name.upper() in secure_headers and
105               hdr_value == secure_headers[hdr_name.upper()]):
106             url_scheme = "https"
107         elif hdr_name == "HOST":
108             server = hdr_value
109         elif hdr_name == "SCRIPT_NAME":
110             script_name = hdr_value
111         elif hdr_name == "CONTENT-TYPE":
112             environ['CONTENT_TYPE'] = hdr_value
113             continue
114         elif hdr_name == "CONTENT-LENGTH":
115             environ['CONTENT_LENGTH'] = hdr_value
116             continue
117 
118         key = 'HTTP_' + hdr_name.replace('-', '_')
119         if key in environ:
120             hdr_value = "%s,%s" % (environ[key], hdr_value)
121         environ[key] = hdr_value
122 
123     environ['wsgi.url_scheme'] = url_scheme
~~~

In my opinion, this is the right way to do it but, for sure, I'd be glad to know other ways.

Regards!
diego

**Comment by alfem** [comment]
Nice, but we do not have it in our Nginx template: https://github.com/gecos-team/gecoscc-installer/blob/master/templates/nginx-gecoscc.conf

**Comment by n1mh** [comment]
That's true, it is not in the template. Should I change the nginx template?

This code works with both http and https connections to GECOSCC and, in case it is installed over HTTPS it will be needed.

**Comment by alfem** [comment]
To check client connection data thru a proxy the standard variable to check is "Forwarded". But currently the 'de-facto' standard is "X-Forwarded-whatever". 

At this moment we should use the last one (until nginx supports the right one). 

Please, include it in the template, and bear in mind both cases: nginx as frontend and also behind a proxy.

**Comment by n1mh** [comment]
> Please, include it in the template, and bear in mind both cases: nginx as frontend and also behind a proxy

Should I keep https configuration for reverse proxy or just the `X-Forwarded_Proto` line?

I'm talking about this:
~~~
proxy_set_header   X-Scheme $scheme;
proxy_set_header   X-SSL-Protocol $ssl_protocol;
proxy_headers_hash_bucket_size 128;
~~~

**Comment by alfem** [comment]
Those are not in our template, are they?

**Pull request merged**